yarn-project/end-to-end/src/e2e_fees/gas_estimation.test.ts

```diff
@@ -95,20 +97,19 @@ describe('e2e_fees gas_estimation', () => {
     const [withEstimate, withoutEstimate] = await sendTransfers(paymentMethod);
     const actualFee = withEstimate.transactionFee!;
 
+    // Actual teardown gas used is less than the limits.
+    expect(estimatedGas.teardownGasLimits.l2Gas).toBeLessThan(gasSettings.teardownGasLimits.l2Gas);
+    expect(estimatedGas.teardownGasLimits.daGas).toBeLessThan(gasSettings.teardownGasLimits.daGas);
+
     // Estimation should yield that teardown has reduced cost, but is not zero
     expect(withEstimate.transactionFee!).toBeLessThan(withoutEstimate.transactionFee!);
-    expect(withEstimate.transactionFee! + teardownFixedFee).toBeGreaterThan(withoutEstimate.transactionFee!);
+    expect(withEstimate.transactionFee!).toBeGreaterThan(withoutEstimate.transactionFee! - teardownFixedFee);
 
     // Check that estimated gas for teardown are not zero since we're doing work there
     expect(estimatedGas.teardownGasLimits.l2Gas).toBeGreaterThan(0);
 
     // Check that the estimate was close to the actual gas used by recomputing the tx fee from it
-    const feeFromEstimatedGas = getFeeFromEstimatedGas(estimatedGas);
-
-    // The actual fee should be under the estimate, but not too much
-    // TODO(palla/gas): 3x is too much, find out why we cannot bring this down to 2x
```

Comment:
It was too much because gasLimits included teardown gas, and `getFeeFromEstimatedGas` doubled the cost for teardown.